packages/@lwc/engine-dom/src/custom-elements/create-custom-element.ts

```diff
@@ -44,56 +55,18 @@ const createUpgradableConstructor = () => {
                 // Do we want to support this? Throw an error? Currently for backwards compat it's a no-op.
             }
         }
-
-        formAssociatedCallback() {
-            const formAssociatedCallback = getLifecycleCallbacks()?.formAssociatedCallback;
-            // if element was upgraded outside LWC, this will be undefined
-            if (!isUndefined(formAssociatedCallback) && !elementsUpgradedOutsideLWC.has(this)) {
-                formAssociatedCallback(this);
-            }
-        }
-
-        formResetCallback() {
-            const formResetCallback = getLifecycleCallbacks()?.formResetCallback;
-            // if element was upgraded outside LWC, this will be undefined
-            if (!isUndefined(formResetCallback) && !elementsUpgradedOutsideLWC.has(this)) {
-                formResetCallback(this);
-            }
-        }
-
-        formDisabledCallback() {
-            const formDisabledCallback = getLifecycleCallbacks()?.formDisabledCallback;
-            // if element was upgraded outside LWC, this will be undefined
-            if (!isUndefined(formDisabledCallback) && !elementsUpgradedOutsideLWC.has(this)) {
-                formDisabledCallback(this);
-            }
-        }
-
-        formStateRestoreCallback() {
-            const formStateRestoreCallback = getLifecycleCallbacks()?.formStateRestoreCallback;
-            // if element was upgraded outside LWC, this will be undefined
-            if (!isUndefined(formStateRestoreCallback) && !elementsUpgradedOutsideLWC.has(this)) {
-                formStateRestoreCallback(this);
-            }
-        }
     }
 
-    // Do not unnecessarily add a connectedCallback/disconnectedCallback, as it introduces perf overhead
+    // Do not unnecessarily add a connectedCallback/disconnectedCallback/etc., as it introduces perf overhead
     // See: https://github.com/salesforce/lwc/pull/3162#issuecomment-1311851174
-    if (hasConnectedCallback) {
-        (UpgradableConstructor.prototype as any).connectedCallback = function () {
-            if (!elementsUpgradedOutsideLWC.has(this)) {
-                connectedCallback(this);
-            }
-        };
-    }
-
-    if (hasDisconnectedCallback) {
-        (UpgradableConstructor.prototype as any).disconnectedCallback = function () {
-            if (!elementsUpgradedOutsideLWC.has(this)) {
-                disconnectedCallback(this);
-            }
-        };
+    if (!isUndefined(lifecycleCallbacks)) {
+        for (const [propName, callback] of entries(lifecycleCallbacks)) {
+            (UpgradableConstructor.prototype as any)[propName] = function () {
+                if (!elementsUpgradedOutsideLWC.has(this)) {
+                    callback(this);
+                }
+            };
+        }
```

Comment:
I also went ahead and DRY'd up this code, although it doesn't turn out to make a perf impact.